### PR TITLE
homectl password caching fixes

### PR DESCRIPTION
**Pull request by poettering**
We were using cached passwords a bit too agressively: once a bad password entered the cache, we'd keep using this even if it turned out to be wrong, never giving the user an option to enter a new, corrected one. Annoying.

### CODE REVIEW

File: src/home/homectl.c [hunk 1]
```diff
@@ -220,6 +220,10 @@ static int acquire_existing_password(
                 return 1;
         }
 
+        /* If this is not our own user, then don't use the password cache */
+        if (is_this_me(user_name) <= 0)
+                flags &= ~(ASK_PASSWORD_ACCEPT_CACHED|ASK_PASSWORD_PUSH_CACHE);
```
Comment (bluca) [inline src/home/homectl.c]: maybe `SET_FLAG(flags, ASK_PASSWORD_ACCEPT_CACHED|ASK_PASSWORD_PUSH_CACHE, false);` ?

**Review by bluca — APPROVED** [review]
LGTM, just one minor nit

File: src/home/homectl.c (hunk 1 above)
Comment (poettering) [reply src/home/homectl.c]: indeed

**Comment by poettering** [comment]
Force pushed new version, addressing @bluca suggestion. No other changes. Upgrading green label since change so minimal.

**Pull request merged**